src/PROPOSAL/detail/PROPOSAL/crosssection/CrossSectionDNDX/CrossSectionDNDXInterpolant.cxx

```diff
@@ -71,7 +71,13 @@ double CrossSectionDNDXInterpolant::Calculate(double energy, double v)
         return 0.;
     auto lim = GetIntegrationLimits(energy);
     v = retransform_v(lim.min, lim.max, v);
-    return interpolant.evaluate(std::array<double, 2> { energy, v });
+    auto dNdx = interpolant.evaluate(std::array<double, 2> { energy, v });
+    if (dNdx < 0) {
+        logger->warn("Negative dNdx value for E = {:.4f} MeV, v = {:.4f} "
```

Comment:
Is it possible to inform the user also about the interaction type?

Reply:
Yes.
The error messages now look like this for example:

```
[2022-04-06 10:16:05.890] [CrossSection.DNDX] [warning] Negative dNdx value for E = 228.6277 MeV, v = 1.0000 detected in interaction type Ioniz. Setting dNdx to zero.
```